apps/studio/src/features/editing-experience/components/form-builder/renderers/controls/JsonFormsImageControl.tsx

```diff
@@ -0,0 +1,128 @@
+import type { ControlProps, RankedTester } from "@jsonforms/core"
+import { useEffect, useState } from "react"
+import { useParams } from "next/navigation"
+import { Box, FormControl, Text } from "@chakra-ui/react"
+import { and, isStringControl, rankWith, schemaMatches } from "@jsonforms/core"
+import { withJsonFormsControlProps } from "@jsonforms/react"
+import {
+  Attachment,
+  FormErrorMessage,
+  FormLabel,
+} from "@opengovsg/design-system-react"
+import wretch from "wretch"
+
+import { JSON_FORMS_RANKING } from "~/constants/formBuilder"
+import { trpc } from "~/utils/trpc"
+import {
+  IMAGE_UPLOAD_ACCEPTED_MIME_TYPES,
+  MAX_IMG_FILE_SIZE_BYTES,
+} from "./constants"
+
+export const jsonFormsImageControlTester: RankedTester = rankWith(
+  JSON_FORMS_RANKING.ImageControl,
+  and(
+    isStringControl,
+    schemaMatches((schema) => schema.format === "image"),
+  ),
+)
+export function JsonFormsImageControl({
+  data,
+  label,
+  handleChange,
+  path,
+  description,
+  required,
+}: ControlProps) {
+  const { pageId, siteId } = useParams()
+
+  const [selectedFile, setSelectedFile] = useState<File | undefined>()
+  const [pendingFile, setPendingFile] = useState<File | undefined>()
+  const [errorMessage, setErrorMessage] = useState("")
+
+  useEffect(() => {
+    if (!!data) {
+      wretch(data as string)
+        .get()
+        .blob()
+        .then((blob) => {
+          const splitData = (data as string).split("/")
+          const fileName = splitData[-1] || "Current Image"
```

Comment:
Filename should be keeping the extension, so when generating the key name for the object, we can use a date-time-page format or something similar which includes the file extension at the back. For the image case, by right it should always have a fileName so it splitData[-1] should always be defined. But anyways, it might not be necessary to have the file extension as part of the file name, from my understanding, these infomation are also stored in the file content(file signature) so it should still render correctly.

Reply:
Filename should be keeping the extension, so when generating the key name for the object, we can use a date-time-page format or something similar which includes the file extension at the back. For the "image" case, by right it should always have a fileName so it splitData[-1] should always be defined. But anyways, it might not be necessary to have the file extension as part of the file name, from my understanding, these infomation are also stored in the file content(file signature) so it should still render correctly.